var/spack/repos/builtin/packages/arrow/package.py

```diff
@@ -146,6 +146,11 @@ def cmake_args(self):
         args.append(self.define_from_variant("ARROW_WITH_ZLIB", "zlib"))
         args.append(self.define_from_variant("ARROW_WITH_ZSTD", "zstd"))
 
+        if not self.spec.satisfies("^re2"):
```

Comment:
This can potentially match indirect deps on re2, i.e. false positives.

Reply:
Thanks for clarifying! I have a feeling there are more patches with this mistake waiting to be upstreamed here, I'll keep an eye out.